Apps/Database/src/Delegates/DbBetaFeatureAccessDelegate.cs

```diff
@@ -57,18 +59,20 @@ public async Task<IList<BetaFeatureAccess>> GetAsync(IEnumerable<string> hdids,
         }
 
         /// <inheritdoc/>
-        public async Task<IList<BetaFeatureAccess>> GetAllAsync(bool includeUserProfile = false, CancellationToken ct = default)
+        public async Task<PaginatedResult<IGrouping<string, BetaFeatureAccess>>> GetAllAsync(int pageIndex, int pageSize, CancellationToken ct = default)
         {
-            IQueryable<BetaFeatureAccess> query = dbContext.BetaFeatureAccess;
-            query = query.OrderBy(x => x.Hdid);
+            ArgumentOutOfRangeException.ThrowIfNegative(pageIndex);
+            ArgumentOutOfRangeException.ThrowIfNegativeOrZero(pageSize);
 
-            if (includeUserProfile)
-            {
-                query = query.Include(p => p.UserProfile);
-                query = query.Where(p => p.UserProfile.Email != null);
-            }
+            IOrderedQueryable<BetaFeatureAccess> query = dbContext.BetaFeatureAccess
+                .Include(p => p.UserProfile)
+                .Where(p => p.UserProfile.Email != null)
+                .OrderBy(p => p.UserProfile.Email);
+
+            int totalCount = await query.CountAsync(ct);
```

Comment:
To avoid 1-3 of 3 issue.  Can you split the queries up?  Get the distant emails first along with count.  Then use the distinct emails to filter the beta access features/

            List<string?> distinctEmails = await dbContext.BetaFeatureAccess
                .Where(p => p.UserProfile.Email != null)
                .Select(p => p.UserProfile.Email)
                .Distinct()
                .ToListAsync(ct);

            IOrderedQueryable<BetaFeatureAccess> query = dbContext.BetaFeatureAccess
                .Include(p => p.UserProfile)
                .Where(p => distinctEmails.Contains(p.UserProfile.Email))
                .OrderBy(p => p.UserProfile.Email);

            int totalCount = distinctEmails.Count;

Reply:
Thanks.  Performing the distinct email query first is a great idea that should fix all the quirks.  And the cost compared to the quirky solution shouldn't be that bad (maybe 1.5x).  I've updated the PR.